yt_dlp/extractor/ondemandkorea.py

```diff
@@ -1,87 +1,170 @@
-            raise ExtractorError(
-                'Unable to access page. You may have been blocked.',
-                expected=True)
-
-        if 'msg_block_01.png' in webpage:
-            self.raise_geo_restricted(
-                msg='This content is not available in your region',
-                countries=self._GEO_COUNTRIES)
-
-        if 'This video is only available to ODK PLUS members.' in webpage:
-            raise ExtractorError(
-                'This video is only available to ODK PLUS members.',
-                expected=True)
-
-        if 'ODK PREMIUM Members Only' in webpage:
-            raise ExtractorError(
-                'This video is only available to ODK PREMIUM members.',
-                expected=True)
-
-        title = self._search_regex(
-            r'class=["\']episode_title["\'][^>]*>([^<]+)',
-            webpage, 'episode_title', fatal=False) or self._og_search_title(webpage)
-
-        jw_config = self._parse_json(
-            self._search_regex((
-                r'(?P<options>{\s*[\'"]tracks[\'"].*?})[)\];]+$',
-                r'playlist\s*=\s*\[(?P<options>.+)];?$',
-                r'odkPlayer\.init.*?(?P<options>{[^;]+}).*?;',
-            ), webpage, 'jw config', flags=re.MULTILINE | re.DOTALL, group='options'),
-            video_id, transform_source=js_to_json)
-        info = self._parse_jwplayer_data(
-            jw_config, video_id, require_title=False, m3u8_id='hls',
-            base_url=url)
-
-        info.update({
-            'title': title,
-            'description': self._og_search_description(webpage),
-            'thumbnail': self._og_search_thumbnail(webpage)
-        })
-        return info
+
+        data = self._download_json(
+            f'https://odkmedia.io/odx/api/v3/playback/{video_id}/', video_id, fatal=False,
+            headers={'service-name': 'odk'}, query={'did': str(uuid.uuid4())}, expected_status=(403, 404))
+        if not traverse_obj(data, ('result', {dict})):
+            msg = traverse_obj(data, ('messages', '__default'), 'title', expected_type=str)
+            raise ExtractorError(msg or 'Got empty response from playback API', expected=True)
+
+        data = data['result']
+
+        def try_geo_bypass(url):
+            return traverse_obj(url, ({parse_qs}, 'stream_url', 0, {url_or_none})) or url
+
+        def try_upgrade_quality(url):
+            mod_url = re.sub(r'_720(p?)\.m3u8', r'_1080\1.m3u8', url)
+            return mod_url if mod_url != url and self._request_webpage(
+                HEADRequest(mod_url), video_id, note='Checking for higher quality format',
+                errnote='No higher quality format found', fatal=False) else url
+
+        potential_urls = traverse_obj(
+            data, (('sources', 'manifest'), ..., 'url', {url_or_none}, {try_geo_bypass}, {try_upgrade_quality}))
+
+        formats = []
+        for url in potential_urls:
+            formats.extend(self._extract_m3u8_formats(url, video_id, fatal=False))
```

Comment:
On second look, I think it'd be better to keep the network requests out of the traversal:
```suggestion
        formats = []
        for m3u8_url in traverse_obj(data, (('sources', 'manifest'), ..., 'url', {url_or_none}, {try_geo_bypass}):
            formats.extend(self._extract_m3u8_formats(try_upgrade_quality(m3u8_url), video_id, fatal=False))
```